packages/client/hmi-client/src/page/project/components/tera-tab-content.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+	<template v-if="assetId && (!isEmpty(tabs) || isDrilldown)">
+		<!--Investigate using component tag since props are similar-->
+		<tera-document
+			v-if="assetType === ProjectAssetTypes.DOCUMENTS"
+			:xdd-uri="getXDDuri(assetId)"
+			:previewLineLimit="10"
+			:project="project"
+			is-editable
+			@open-code="openCode"
+			@asset-loaded="emit('asset-loaded')"
+		/>
+		<tera-model
+			v-else-if="assetType === ProjectAssetTypes.MODELS"
+			:asset-id="assetId"
+			:project="project"
+			is-editable
+			@asset-loaded="emit('asset-loaded')"
+		/>
+		<tera-dataset
+			v-else-if="assetType === ProjectAssetTypes.DATASETS"
+			:asset-id="assetId"
+			:project="project"
+			is-editable
+			@asset-loaded="emit('asset-loaded')"
+		/>
+		<simulation-plan
+			v-else-if="assetType === ProjectAssetTypes.PLANS"
+			:asset-id="assetId"
+			:project="project"
+			@asset-loaded="emit('asset-loaded')"
+		/>
+		<simulation-run
+			v-else-if="assetType === ProjectAssetTypes.SIMULATION_RUNS"
+			:asset-id="assetId"
+			:project="project"
+			@asset-loaded="emit('asset-loaded')"
+		/>
+	</template>
+	<code-editor
+		v-else-if="assetType === ProjectAssetTypes.CODE"
+		:initial-code="code"
+		@on-model-created="openNewModelFromCode"
+	/>
+	<tera-model
```

Comment:
Why are there two `tera-models`? Is one for an existing model and one for a new one?

Reply:
Yeah the one on the outside is for when you're creating one.

Reply:
Just wondering if there is any value to consolidating them into the same component. Intuitively I would think `tera-model` would display either a new model or saved model instead needing two instances of `tera-model`.